src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectAssetFileWatcher.cs

```diff
@@ -185,11 +203,21 @@ private IProjectTree FindProjectJsonNode(IProjectTree newTree, string projectFil
             return null;
         }
 
-        private void RegisterFileWatcherAsync(string projectLockJsonFilePath)
+        private void RegisterFileWatcher(string projectLockJsonFilePath)
         {
             // Note file change service is free-threaded
             if (_fileChangeService != null && projectLockJsonFilePath != null)
             {
+                if (File.Exists(projectLockJsonFilePath))
+                {
+                    _previousContents = File.ReadAllText(projectLockJsonFilePath);
```

Comment:
These files tend to be really large - in a large solution, this is going to be a significant amount of memory.  If we need to do something like this, we should calculate the hash and store that instead of the entire contents.

Reply:
Would `String.GetHashCode` be acceptable here? Or do I need to use a different hash method?